src/cascadia/TerminalApp/CascadiaSettingsSerialization.cpp

```diff
@@ -633,17 +635,14 @@ void CascadiaSettings::_LayerOrCreateProfile(const Json::Value& profileJson)
 // Return Value:
 // - a Profile that can be layered with the given json object, iff such a
 //   profile exists.
-Profile* CascadiaSettings::_FindMatchingProfile(const Json::Value& profileJson)
+winrt::com_ptr<implementation::Profile> CascadiaSettings::_FindMatchingProfile(const Json::Value& profileJson)
 {
     for (auto& profile : _profiles)
     {
-        if (profile.ShouldBeLayered(profileJson))
+        auto profileImpl = winrt::get_self<winrt::TerminalApp::implementation::Profile>(profile);
+        if (profileImpl->ShouldBeLayered(profileJson))
         {
-            // HERE BE DRAGONS: Returning a pointer to a type in the vector is
-            // maybe not the _safest_ thing, but we have a mind to make Profile
-            // and ColorScheme winrt types in the future, so this will be safer
-            // then.
-            return &profile;
+            return profileImpl->get_strong();

```

Comment:
wait isn't `profileImpl` already a strong `com_ptr`?

Reply:
Apparently `winrt::get_self<D>` returns `D*` instead of `winrt::com_ptr<D>`. I don't get why `D*` can't be auto-converted to a smart pointer. Ugh!

Reply:
oh okay, that makes sense (and maybe explains an issue I was having earlier today)